app/webhook.go

```diff
@@ -323,6 +340,21 @@ func (a *App) CreateIncomingWebhookForChannel(creatorId string, channel *model.C
 		return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_username.app_error", nil, "", http.StatusBadRequest)
 	}
 
+	if len(hook.WhiteIpList) > 0 {
+		if !(model.IsValidIpList(hook.WhiteIpList)) {
+			return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_white_ip_list.app_error", nil, "", http.StatusBadRequest)
+		}
+	} else if len(hook.HmacAlgorithm) > 0 {
+		if hook.IsValidModels() {
+			hook.SecretToken = model.NewRandomString(26)
+			// hook.SecretToken = "me" // for testing
+		} else {
+			return nil, model.NewAppError("CreateIncomingWebhookForChannel", "api.incoming_webhook.invalid_signing_model.app_error", nil, "", http.StatusBadRequest)
+		}
+	} else {
+		hook.ResetModels()
```

Comment:
`ResetModels` is opaque to me in this context -- what are "models"?

Reply:
Indeed. I renamed it to ResetSignatureHeaserModels. This is just for cleaning the request from a channel admin/owner configures his incoming webook endpoint. If algorithm is not selected/chosen, reset all the signature models to empty. There is a case for example someone uses curl to post some data in the signature models while leave empty for algorithm. I did the reset in whitelist as well.